Finish overall server part , including private & public room

jason810496 opened the pull request:
( I haven't test the functionality 

implementation of the `play` function will be finished after the game logic is done.

jason810496 committed:
Finish overall server part , including private & public room

( haven't test the functionality

Ibrahim2750mi committed:
fixed linting and cleaned up

Ibrahim2750mi committed:
fixed linting 2

Ibrahim2750mi reviewed (approved):
Made some changes to your code(not big), just fixed the styling and some other things have a look, if its ok, merge it then.

Ibrahim2750mi commented:
linked with issue #5

jason810496 committed:
update server & client

server : finish private room & public connection mechanism

client : able to demo connection

Todo :

wait untill game logic is finished then combine game class in both file

jason810496 committed:
fix lint

jason810496 committed:
lint again

jason810496 committed:
lint for the third time

jason810496 committed:
lint finish

Ibrahim2750mi committed:
final cleaning up

Ibrahim2750mi committed:
fixed linting

Ibrahim2750mi committed:
fixed linting again
